src/server/debug/server/server.go

```diff
@@ -927,3 +952,123 @@ func redirectDump(ctx context.Context, c debug.DebugClient, filter *debug.Filter
+	return s.collectDatabaseQuery(ctx, tw, "row-counts", queryDatabaseRowCounts, prefix...)
+}
+
+func (s *debugServer) collectDatabaseTableSizes(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "table-sizes", queryDatabaseTableSizes, prefix...)
+}
+
+func (s *debugServer) collectDatabaseQuery(ctx context.Context, tw *tar.Writer, fileName string,
+	queryFunc func(conn *sql.Conn, ctx context.Context, w io.Writer) error, prefix ...string) error {
+	return collectDebugFile(tw, fileName, "txt", func(w io.Writer) error {
+		conn, err := s.env.GetDBClient().Conn(ctx)
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		defer conn.Close()
+		if err = queryFunc(conn, ctx, w); err != nil {
+			return errors.EnsureStack(err)
+		}
+		return errors.EnsureStack(err)
+	}, prefix...)
+}
+
+func queryDatabaseRowCounts(conn *sql.Conn, ctx context.Context, w io.Writer) error {
+	query := `
+		SELECT schemaname, relname, n_live_tup, seq_scan, idx_scan 
+		  FROM pg_stat_user_tables 
+		ORDER BY relname DESC;
+	`
+	return queryDatabase(conn, ctx, query, w)
+}
+
+func queryDatabaseActivities(conn *sql.Conn, ctx context.Context, w io.Writer) error {
+	query := `
+		SELECT current_timestamp - query_start as runtime, datname, usename, client_addr, query
+		FROM pg_stat_activity
+		WHERE state != 'idle'
+		ORDER by runtime desc
+		LIMIT 100;
+	`
+	return queryDatabase(conn, ctx, query, w)
+}
+
+func queryDatabaseTableSizes(conn *sql.Conn, ctx context.Context, w io.Writer) error {
+	query := `
+		SELECT nspname || '.' || relname AS "relation",
+		pg_size_pretty(pg_total_relation_size(C.oid)) AS "total_size"
+		FROM pg_class C
+		LEFT JOIN pg_namespace N ON (N.oid = C.relnamespace)
+		WHERE nspname NOT IN ('pg_catalog', 'information_schema')
+		  AND C.relkind <> 'i'
+		  AND nspname !~ '^pg_toast'
+		ORDER BY pg_total_relation_size(C.oid) DESC
+	`
+	return queryDatabase(conn, ctx, query, w)
+}
+
+func queryDatabase(conn *sql.Conn, ctx context.Context, query string, w io.Writer) error {
+	rows, err := conn.QueryContext(ctx, query)
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	return writeRows(rows, w)
+}
+
+func writeRows(rows *sql.Rows, w io.Writer) error {
+	cols, err := rows.Columns()
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	header := strings.Join(cols, "\t") + "\n"
+	tabWriter := tabwriter.NewWriter(w, header)
+	for rows.Next() {
+		err = writeRowToTabWriter(rows, tabWriter)
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+	}
+	if err = tabWriter.Flush(); err != nil {
+		return errors.EnsureStack(err)
+	}
+	return nil
+}
+
+func writeRowToTabWriter(rows *sql.Rows, tabWriter *tabwriter.Writer) error {
+	results := make(map[string]interface{})
+	if err := sqlx.MapScan(rows, results); err != nil {
+		return errors.EnsureStack(err)
+	}
+	row := ""
+	for _, value := range results {
```

Comment:
I don't think this will print rows in the right order. Maybe SliceScan will?

Reply:
columns in the right order, I think you mean.  You can SliceScan or pass in rows.Columns() and iterate over that instead of the map keys.

I think the rows should just be JSON objects, though.  JSONlines format files are easy to analyze with JQ.  The first thing I do with a CSV file is convert it to JSON anyway ;)